### DISCUSSION TITLE
Unsolicited Messages

### DISCUSSION THREAD

**Discussion by romainrobotics** [ottowayi/pycomm3#252] [Q&A] [answered]
I've found that this library can establish a connection to a PLC, can it do the opposite? i.e. can this be used to allow a PLC to establish a connection, then send unsolicited generic messages? 

For example, my PLC sends a "list services" request, I'd like to use a library to respond to it but I don't see that as an option anywhere. Is that outside the scope?

**Comment by ottowayi** [comment, marked as answer]
Currently that functionality is not supported, this library only acts as a client.  But, server functionality is something I would like to eventually add.  I'm in the process of rewriting almost the entire library and the new way I'm doing it should make this possible.  Essentially every object will be able to be round-tripped, ie both encodable from a python object to bytes and from bytes back into a python object.  The current client functionality encodes requests and decodes responses, to write a server it just needs to be able to decode requests and encode responses, which will essentially be provided by default.  I've been working on this for a while now, but I haven't had a lot of time to dedicate to it.  So, I can't promise a timeline or anything.  Until then, I think `cpppo` may be able to help you, but it's been a couple years since I last looked at it.
